browser/js/dashboard/gridbox/widgetCtrl.js

```diff
@@ -18,5 +20,16 @@ app.controller('WidgetCtrl', ['$scope', '$uibModal',
         })
       };
 
-    }
-])
\ No newline at end of file
+  $scope.updateData = function(widget){
+    if (widget.refreshInterval > 0) {
+        $interval(function(){
+          return WidgetSettingsFactory.newSetKeys(widget.dataSource)
```

Comment:
Don't think the `return` here is necessary, not sure what `$interval` would do with it.